dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -703,17 +742,61 @@ public void completeFile(String path, CompleteFilePOptions options, String uuid)
       handle.close();
       handle = null; // no more use of this handle
 
-      // The simplest way of updating metadata is invalidating cache in worker.
-      // Next time, worker will get fresh metadata from ufs.
-      AlluxioURI fullPathUri = new AlluxioURI(path);
-      AlluxioURI parentDir;
-      if (fullPathUri.isRoot()) {
-        parentDir = fullPathUri;
+      invalidateFileMeta(path);
+    }
+  }
+
+  @Override
+  public void delete(String path, DeletePOptions options) {
+    try {
+      invalidateFileMeta(path);
+      invalidateCachedFile(path);
+
+      // TODO(hua) Close the open file handle?
+
+      UfsStatus status = mUfs.getStatus(path);
+      if (status.isFile()) {
+        mUfs.deleteFile(path);
```

Comment:
Should we delete the file in UFS first? This is because we can roll back easily when UFS operation fails. But if we delete the file in Alluxio first, we may still encounter exception during the time we delete the file in UFS. In that case, we have to deal with the roll back logic.

Reply:
Thank you for raising this question.
The following two statements only invalidate the cached metadata and data in PagingStore.
If something wrong happens, we can reload the metadata and data from UFS.
So the source of truth is UFS.

I would prefer to the current implementation.